lib/mrsk/cli/server.rb

```diff
@@ -1,17 +1,14 @@
 class Mrsk::Cli::Server < Mrsk::Cli::Base
-  desc "bootstrap", "Ensure curl and Docker are installed on servers"
+  desc "bootstrap", "Ensure readiness to run MRSK"
   def bootstrap
-    with_lock do
-      on(MRSK.hosts + MRSK.accessory_hosts) do
-        dependencies_to_install = Array.new.tap do |dependencies|
-          dependencies << "curl" unless execute "which curl", raise_on_non_zero_exit: false
-          dependencies << "docker.io" unless execute "which docker", raise_on_non_zero_exit: false
-        end
+    unready = []
 
-        if dependencies_to_install.any?
-          execute "apt-get update -y && apt-get install #{dependencies_to_install.join(" ")} -y"
-        end
-      end
+    on(MRSK.hosts + MRSK.accessory_hosts) do |host|
+      unready << host unless execute(:docker, :info, raise_on_non_zero_exit: false)
+    end
+
+    if unready.any?
+      raise "Docker is not running on #{unready.join(", ")}. Install it first: https://docs.docker.com/engine/install/"
```

Comment:
Indicate how to install Docker rather than automating it. That only worked with SSH root users—which most hosts strongly discourage—and on Debian/Ubuntu hosts. Clearer to indicate what needs to be set up. Allows MRSK to remain indifferent to deploy target platform.

Reply:
Hmm, Digital Ocean and Hetzner are two examples where you can just use the root user to do the install. I don't mind that we catch missing docker better, but it seems like a step backwards to me without an upside to not automate this step when possible.

Reply:
IMHO, the option proposed by @jeremy is better, otherwise mrsk has to detect if the user is root and implement different installation methods to support systems based on apt, yum, etc. which complicates the logic a lot. And in the case of ubuntu, the recommended option is to use the binaries from the docker repository instead of the ones that ubuntu brings by default in its repositories. If you want to automate the docker installation process when creating the instance, you can always use cloud-init to provision it.

Reply:
Updated to attempt an auto-install for root users only using the [multi-OS install script](https://get.docker.com).

It's not foolproof, but it relieves MRSK of trying to drive multiple OS and packaging systems.